src/scenes/NftDetailPage/NftAdditionalDetails.tsx

```diff
@@ -27,28 +29,36 @@ function NftAdditionalDetails({ nft }: Props) {
       <Spacer height={12} />
       {showAdditionalDetails && (
         <div>
-          <BodyRegular color={colors.gray50}>Contract address</BodyRegular>
-          <StyledLink
-            href={`https://etherscan.io/address/${nft.asset_contract.address}`}
-            target="_blank"
-            rel="noreferrer"
-          >
-            <BodyRegular>{nft.asset_contract.address}</BodyRegular>
-          </StyledLink>
+          {nft.asset_contract.address && nft.asset_contract.address !== '' && (
```

Comment:
since this logic is used a couple times, you can define a variable in this component that's like:

```js
const hasContractAddress = nft.asset_contract?.address !== ''

...

{hasContractAddress && <render>}
```

notice the optional chain also helps reduce a bit of boilerplate

Reply:
big brain ty